packages/manager/modules/telecom-universe-components/src/telecom/pack/migration/pack-migration-process.service.js

```diff
@@ -296,13 +290,25 @@ export default /* @ngInject */ function($q, OvhApiPackXdsl, Poller) {
                 })),
               );
             });
-            set(offer, 'options', keyBy(offer.options, 'name'));
+            const voipOptions = {};
+            offer.options
+              .filter((option) => option.name === OPTION_NAME)
+              .map((option) => {
+                return { [`${option.name}_${option.optional}`]: option };
+              })
+              .forEach((option) => {
+                Object.assign(voipOptions, option);
+              });
```

Comment:
**suggestion:** `map` and `forEach` can be combined to `reduce` and avoid multiple looping.
```suggestion
            const voipOptions = offer.options
              .filter((option) => option.name === OPTION_NAME)
              .reduce((acc, option) => {
                set(acc, `${option.name}_${option.optional}`, option);
                return acc;
              }, {});
```

Reply:
thank you, change made